ember/app/abilities/-base.js

```diff
@@ -35,4 +35,11 @@ export default class BaseAbility extends Ability {
   isStaffOrOwnIdentity(identity) {
     return this.isStaff || this.isOwnIdentity(identity);
   }
+
+  hasAccess(document) {
+    return !!document.accesses.findBy(
+      "identity.idpId",
+      this.session.data.authenticated.userinfo.sub
+    );
```

Comment:
Type casting can also be done like this. IMO easier readable
```suggestion
    return Boolean(document.accesses.findBy(
      "identity.idpId",
      this.session.data.authenticated.userinfo.sub
    ));
```

Reply:
Changed